src/wallet/coinselection.h

```diff
@@ -102,26 +138,27 @@ struct OutputGroup
     CAmount effective_value{0};
     /** The fee to spend these UTXOs at the effective feerate. */
     CAmount fee{0};
-    /** The target feerate of the transaction we're trying to build. */
-    CFeeRate m_effective_feerate{0};
     /** The fee to spend these UTXOs at the long term feerate. */
     CAmount long_term_fee{0};
-    /** The feerate for spending a created change output eventually (i.e. not urgently, and thus at
-     * a lower feerate). Calculated using long term fee estimate. This is used to decide whether
-     * it could be economical to create a change output. */
-    CFeeRate m_long_term_feerate{0};
+    /** Coin selection parameters relevant to this OutputGroup.
+     * The particular parameters used are the effective feerate (to calculate the effective value and fee),
+     * the long term feerate (to calculate the long term fee), and whether the fee is to be subtracted from the outputs.
+     */
+    CoinSelectionParams m_cs_params;
 
     OutputGroup() {}
-    OutputGroup(const CFeeRate& effective_feerate, const CFeeRate& long_term_feerate) :
-        m_effective_feerate(effective_feerate),
-        m_long_term_feerate(long_term_feerate)
+    OutputGroup(const CoinSelectionParams& params) :
+        m_cs_params(params)
     {}
 
     void Insert(const CInputCoin& output, int depth, bool from_me, size_t ancestors, size_t descendants, bool positive_only);
     bool EligibleForSpending(const CoinEligibilityFilter& eligibility_filter) const;
+    CAmount GetSelectionAmount() const;
 };
 
-bool SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& target_value, const CAmount& cost_of_change, std::set<CInputCoin>& out_set, CAmount& value_ret, CAmount not_input_fees);
+bool SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& selection_target, const CAmount& cost_of_change, std::set<CInputCoin>& out_set, CAmount& value_ret);
+
+bool SelectCoinsSRD(std::vector<OutputGroup>& utxo_pool, const CAmount& target_value, std::set<CInputCoin>& out_set, CAmount& value_ret);
```

Comment:
Please add doxygen comments for new functions.

Reply:
Done